FIX: CI Linting

ukch opened the pull request:
* delete an unused module
* code to be run in the browser should now have a `.browser.js` extension, so it is linted correctly

ukch committed:
Fix: add yaml files to lint-staged config

ukch committed:
Fix all eslint errors

Fydon commented on src/routes/search_routes.js:
```diff
@@ -93,7 +93,7 @@ module.exports = function (app, ensureAuthenticatedAndCheckIDP, log) {
         util.dedup(
           searchStr
-            .replace(/[- .,;–—"“”'‘’`~!()|\\{}\[\]:<>/?*]+/g, " ")
+            .replace(/[- .,;–—"“”'‘’`~!()|\\{}[\]:<>/?*]+/g, " ")
```
Why is the escape for `[` being removed?

ukch replied:
According to eslint, it is not necessary to escape the `[` character in this context.

Fydon commented on src/routes/user_routes.js:
```diff
@@ -59,15 +59,15 @@ module.exports = function (app, ensureAuthenticatedAndCheckIDP, ensureAuthentica
     const locale = language || req.idpLang || 'en'
 
-    const urlWithoutEditing = `${util.getProtocol({ req })}://${req.headers.host}${req.originalUrl.replace(/([\?&])editing=1&?/, '$1').replace(/iniframe=1&?/, '').replace(/[\?&]$/, '')}`
+    const urlWithoutEditing = `${util.getProtocol({ req })}://${req.headers.host}${req.originalUrl.replace(/([?&])editing=1&?/, '$1').replace(/iniframe=1&?/, '').replace(/[?&]$/, '')}`
```
Why are the escapes for `?` being removed?

ukch replied:
According to eslint, it is not necessary to escape the `?` character in this context.

Fydon commented on src/utils/util.js:
```diff
@@ -214,7 +214,7 @@ const util = {
 
     // Split timestamp into [ Y, M, D, h, m, s, ms ]
-    var t = mysqlDatetime.split(/[- :\.]/)
+    var t = mysqlDatetime.split(/[- :.]/)
```
Why is the escape for `.` being removed?

ukch replied:
According to eslint, it is not necessary to escape the `.` character in this context.

Fydon commented on src/utils/util.js:
```diff
@@ -1347,15 +1349,15 @@ const util = {
 
   isValidEmail: email => {
-    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Why are the escapes for `[` being removed?

Fydon reviewed (changes requested):
Thank you for taking the time to deal with all these issues.

`widget_setup.js‎` must be available at `scripts/widget_setup.js‎` for existing embeds of the ereader to continue to work.

ukch commented:
> `widget_setup.js‎` must be available at `scripts/widget_setup.js‎` for existing embeds of the ereader to continue to work.

~~Although this file has been renamed at source, it is still available at the same URL, that is "src/js/widget_setup.js". Any existing embed will still be able to find the file at the same URL as it was available in the past.~~

Edit: I was misunderstanding what this route was doing. I have since refactored it to be more sensible.

Fydon commented:
The embed for staging is at `https://stg.read.biblemesh.com/scripts/widget_setup.js`

<img width="3833" height="1952" alt="image" src="https://github.com/user-attachments/assets/16045d2f-4f8f-4a36-b666-c6a25e17d2b6" />

ukch commented:
I'm seeing this stack trace when running the Docker image:
```
reader-1                | Error: Cannot find module '../utils/web-funcs'
reader-1                | Require stack:
reader-1                | - /app/src/routes/auth_routes.js
reader-1                | - /app/src/routes/routes.js
reader-1                | - /app/app.js
reader-1                |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
reader-1                |     at Function.Module._load (node:internal/modules/cjs/loader:873:27)
reader-1                |     at Module.require (node:internal/modules/cjs/loader:1100:19)
reader-1                |     at Module.patchedRequire (/app/node_modules/require-in-the-middle/index.js:233:34)
reader-1                |     at Module.Hook._require.Module.require (/app/node_modules/require-in-the-middle/index.js:181:27)
reader-1                |     at Module.patchedRequire (/app/node_modules/require-in-the-middle/index.js:233:34)
reader-1                |     at Module.Hook._require.Module.require (/app/node_modules/require-in-the-middle/index.js:181:27)
reader-1                |     at require (node:internal/modules/cjs/helpers:119:18)
reader-1                |     at Object.<anonymous> (/app/src/routes/auth_routes.js:5:39)
reader-1                |     at Module._compile (node:internal/modules/cjs/loader:1198:14) {
reader-1                |   code: 'MODULE_NOT_FOUND',
reader-1                |   requireStack: [
reader-1                |     '/app/src/routes/auth_routes.js',
reader-1                |     '/app/src/routes/routes.js',
reader-1                |     '/app/app.js'
reader-1                |   ]
reader-1                | }
```
To investigate

ukch commented on src/routes/auth_routes.js:
```diff
@@ -2,6 +2,7 @@ const { log } = require('../utils/logger')
 const util = require('../utils/util')
 const { i18n } = require("inline-i18n")
 const sendEmail = require("../utils/sendEmail")
+const { makePostStatusToParentStr } = require('../utils/web-funcs');
```
I guess I should change this:
```suggestion
const { makePostStatusToParentStr } = require('../utils/web-funcs.browser');
```
To investigate

ukch committed:
Fix import path

ukch committed:
Serve the correct file; add a test

ukch commented:
> `widget_setup.js‎` must be available at `scripts/widget_setup.js‎` for existing embeds of the ereader to continue to work.

I fixed this and added a test. Please re-review.

Fydon reviewed (approved):
Thank you. If you've validated that the regex checks haven't changed, I'm happy with these changes
